### PR TITLE
okhttp: better reporting for responses

### PR DESCRIPTION
**Pull request by mgenov**
The okhttp client now reports errors betters then now.

### CODE REVIEW

**Comment by mgenov** [comment]
After this PR is merged the maven repository will be updated with the latest jars.

File: jwt-java-client-okhttp/src/test/java/com/clouway/oauth2/client/IssueJwtTokensTest.java [hunk 1]
```diff
@@ -84,19 +85,27 @@ public void issueNewToken() throws IOException, NoSuchAlgorithmException, Invali
...
+      );
+
+      JwtConfig config = new Builder("2000001@apps.telcongserviceaccount.com", clientDriver.getBaseUrl() + "/o/oauth2/v1/token", PRIVATE_KEY.getBytes())
+              .subject("someuser@clouway.com")
+              .build();
+
+      TokenSource tokenSource = config.tokenSource();
+      tokenSource.token(new Date());
+      
+      fail("IOException was not thrown when token endpoint fails?");
```
Comment (VasilMitovClouway) [inline jwt-java-client-okhttp/src/test/java/com/clouway/oauth2/client/IssueJwtTokensTest.java]: Message is misleading for me. I think the question mark is the reason. I think just "IOException was not thrown on token endpoint failure." sounds better

**Review by VasilMitovClouway — APPROVED** [review]
LGTM after fix comments.

**Comment by VasilMitovClouway** [comment]
Does the previous comment about the update with the latest jars include the oauth2-server jar as well ?

File: jwt-java-client-okhttp/src/test/java/com/clouway/oauth2/client/IssueJwtTokensTest.java (hunk 1 above)
Comment (mgenov) [reply jwt-java-client-okhttp/src/test/java/com/clouway/oauth2/client/IssueJwtTokensTest.java]: Change is done. Waiting for final review.

**Comment by mgenov** [comment]
No, projects are different. I'm releasing oauth2-server in a few minutes.

**Comment by mgenov** [comment]
"LGTM after fix comments." -> don't do this one review. It's bad habit.

**Comment by mgenov** [comment]
The release is already available in the sonatype repository. In an hour it should be and in maven central. 
 
https://oss.sonatype.org/content/repositories/releases/com/clouway/security/oauth2-server/1.0-RC21/

**Pull request merged**